packages/ai-jsx/src/lib/openai.tsx

```diff
@@ -437,7 +437,7 @@ export async function* OpenAIChatModel(
       if (props.experimental_streamFunctionCallOnly) {
         yield JSON.stringify({
           ...currentMessage.function_call,
-          arguments: untruncateJson.default(currentMessage.function_call.arguments ?? '{}'),
+          arguments: JSON.parse(untruncateJson.default(currentMessage.function_call.arguments || '{}')),
```

Comment:
ah, was this the problem with my "sometimes it's a multiply-nested string" issue?

Reply:
Yup exactly. Should be good now.